test/Microsoft.VisualStudio.Threading.Tests/JoinableTaskContextTests.cs

```diff
@@ -45,6 +45,41 @@ public void IsWithinJoinableTask()
         });
     }
 
+    [Fact]
+    public void IsMainThreadBlockedByAnyoneTrue()
+    {
+        Assert.False(this.Context.IsMainThreadBlockedByAnyone);
+        this.Factory.RunAsync(async delegate
```

Comment:
Your test will never fail due to exceptions thrown from the RunAsync delegate, since you're dropping the result. In fact, the test will complete before the delegate has even fully executed. 
Did you mean to use `this.Factory.Run` here?

Reply:
Test was updated. I had thought inside JTF.Run is the same as inside WaitSynchronouslyCore.